src/game/client/components/menus.cpp

```diff
@@ -2251,25 +2241,18 @@ void CMenus::OnRender()
 	float MouseY = (m_MousePos.y/(float)Graphics()->ScreenHeight())*pScreen->h;
 	UI()->Update(MouseX, MouseY, MouseX*3.0f, MouseY*3.0f);
 
-	// render
-	if(Client()->State() != IClient::STATE_DEMOPLAYBACK)
-		Render();
+	// render menu
+	Graphics()->MapScreen(pScreen->x, pScreen->y, pScreen->w, pScreen->h);
+	if(Client()->State() == IClient::STATE_DEMOPLAYBACK)
+		RenderDemoPlayer(*pScreen);
```

Comment:
There's a return some lines above so this function might not be called. F.e. when the demo player menu is collapsed the bar doesn't show anymore.

Reply:
I restructured it so there is only one return at the and the `xyzPressed = false` at the end can be reused.

At first I wasn't sure if `UI()->FinishCheck()` should also be called if the menu is inactive, as this would reset the active item when the menu becomes inactive. It's implemented like it way before now, so the active item stays active when the menu becomes inactive.